hi-ml-azure/src/health_azure/datasets.py

```diff
@@ -152,6 +152,9 @@ def _get_latest_v2_asset_version(ml_client: MLClient, data_asset_name: str) -> s
         if data_asset_version > highest_version:
             highest_version = data_asset_version
 
+    if highest_version == 0:
+        raise ResourceNotFoundError(f"No data asset found with the provided name: {data_asset_name}")
```

Comment:
Nitpick - this could strictly speaking also happen if one of the data_asset.version == "0", might be safer to work with None as the default?